controllers/statussync/policy_status_sync.go

```diff
@@ -441,7 +490,225 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 	return reconcile.Result{}, nil
 }
 
+func parseComplianceFromMessage(message string) policiesv1.ComplianceState {
+	cleanMsg := strings.ToLower(
+		strings.TrimSpace(
+			strings.TrimPrefix(message, "(combined from similar events):"),
+		),
+	)
+
+	if strings.HasPrefix(cleanMsg, "compliant") {
+		return policiesv1.Compliant
+	} else if strings.HasPrefix(cleanMsg, "pending") {
+		return policiesv1.Pending
+	}
+
+	return policiesv1.NonCompliant
+}
+
+// ceRequestFromEvent converts a Kubernetes event relating to a policy controller's compliance event to a
+// struct representing a POST request to compliance events API. An error is returned if the event can't be converted.
+func ceRequestFromEvent(event *corev1.Event) (utils.ComplianceAPIEventRequest, error) {
+	ce := utils.ComplianceAPIEventRequest{UID: event.UID}
+
+	pID, err := strconv.ParseInt(event.Annotations[utils.PolicyDBIDAnnotation], 10, 32)
+	if err != nil {
+		log.Error(err, "The event had an invalid policy ID", "policyID", event.Annotations[utils.PolicyDBIDAnnotation])
+
+		return ce, fmt.Errorf("the event had an invalid policy ID: %s", event.Annotations[utils.PolicyDBIDAnnotation])
+	}
+
+	ce.Policy = utils.ComplianceAPIEventPolicyID{ID: int32(pID)}
+
+	if event.Annotations[utils.ParentDBIDAnnotation] != "" {
+		// The parent policy ID is optional so continue even if it's invalid
+		ppID, err := strconv.ParseInt(event.Annotations[utils.ParentDBIDAnnotation], 10, 32)
+		if err == nil {
+			ce.ParentPolicy = &utils.ComplianceAPIEventPolicyID{ID: int32(ppID)}
+		}
+	}
+
+	compliance := parseComplianceFromMessage(event.Message)
+
+	ce.Event = utils.ComplianceAPIEvent{
+		Compliance: compliance,
+		Message:    strings.TrimLeft(event.Message[len(compliance):], " ;"),
+		Timestamp:  event.LastTimestamp.Format(time.RFC3339Nano),
+		ReportedBy: "governance-policy-framework",
+	}
+
+	return ce, nil
+}
+
+// StartComplianceEventsSyncer will monitor the events queue and record compliance events on the compliance events
+// API. It uses either certificate or token authentication to authenticate with the compliance events API using the
+// configuration in hubCfg. Note that apiURL is the base URL to the API. It should not contain the path to the
```

Comment:
Right, we're using the API Server to talk to the hub not the default ingress.  It probably still needs to use reencrypt since the propagator should be expecting to see tls with the service CA not the api server ca.
I'm guessing there is no token auth option here anymore too.

Reply:
Good point. I just changed it to use the system root CAs + the API server CA to account for both situations.